pkg/manager/member/pd_member_manager.go

```diff
@@ -238,7 +238,7 @@ func (m *pdMemberManager) syncPDStatefulSetForTidbCluster(tc *v1alpha1.TidbClust
 		return err
 	}
 
-	if m.deps.CLIConfig.AutoFailover {
+	if m.deps.CLIConfig.AutoFailover && tc.Spec.PD.MaxFailoverCount != nil && *tc.Spec.PD.MaxFailoverCount > 0 {
```

Comment:
What happens if failover occurs and set the max failover count to 0?

Reply:
can not arrive the `Recover` func?

Reply:
The failover Pods will not be scaled in and the user needs to remove the failure member manually in the status.pd then the failover Pods will be scaled in.

Reply:
@hoyhbx After checking the logic of the other component, I think @KanShiori 's above comment is reasonable, we should be able to recover the previous failover Pods even if we set MaxFailoverCount to 0.

Reply:
Could you please add the check in L244?

Reply:
Sure, I have changed it. @DanielZhangQD